test/components/practice.spec.coffee

```diff
@@ -0,0 +1,170 @@
+{expect} = require 'chai'
+_ = require 'underscore'
+React = require 'react/addons'
+Router = require 'react-router'
+{Promise} = require('es6-promise')
+
+{CourseActions, CourseStore} = require '../../src/flux/course'
+{TaskActions, TaskStore} = require '../../src/flux/task'
+{TaskStepActions, TaskStepStore} = require '../../src/flux/task-step'
+
+{SinglePractice, Tasks} = require '../../src/components'
+{routes} = require '../../src/router'
+
+VALID_MODEL = require '../../api/courses/1/practice.json'
+
+
+div = document.createElement('div')
+
+routerHelper = (route) ->
+  history = new Router.TestLocation([route])
+  promise = new Promise (resolve, reject) ->
+    Router.run routes, history, (Handler, state)->
+      router = @
+      React.render(<Handler/>, div, ()->
+        component = @
+        result = {div, component, state, router, history}
+        resolve(result)
+      )
+
+tasksHelper = (courseId) ->
+  routerHelper("/courses/#{courseId}/tasks")
+
+courseHelper = (model, courseId) ->
+  # Load practice in CourseStore
+  CourseActions.loaded(model, courseId)
+  routerHelper("/courses/#{courseId}/practice")
+
+
+
+exerciseTestActions = 
+  fillFreeResponse: (node) ->
+    textarea = node.querySelector('textarea')
+    textarea.value = 'Test Response'
+    React.addons.TestUtils.Simulate.focus(textarea)
+    React.addons.TestUtils.Simulate.keyDown(textarea, {key: 'Enter'})
+    React.addons.TestUtils.Simulate.change(textarea)
+
+    textarea
+
+  clickContinueButton: (node) ->
+    continueButton = node.querySelector('button.btn-primary')
+    React.addons.TestUtils.Simulate.click(continueButton)
+    continueButton = node.querySelector('button.btn-primary')
+
+
+
+describe 'Practice Widget', ->
+  beforeEach ->
+    CourseActions.reset()
+    TaskActions.reset()
+    TaskStepActions.reset()
+
+  afterEach ->
+    React.unmountComponentAtNode(div)
+
+  it 'should load the practice button on the course tasks page', (done) ->
+    tests = ({div}) ->
+      expect(div.querySelector('.-practice')).to.not.be.null
+      done()
+
+    tasksHelper(1).then(tests).catch(done)
```

Comment:
is catching the error and calling done correct? I would have expected something like `.catch(fail)` from `{expect, fail} = require 'chai'` (not sure if that's the exact syntax) or just dropping the `.catch` and letting the test time out.